### PR TITLE
[FIX] address missing OpenMS.ini problem

### PR DESCRIPTION
**Pull request by hroest**
attempt to fix OS X failures on travis

addresses issue #1924 by catching a failure to read the OpenMS.ini file and not caring
potential improvement: only do this if the -test flag is set

### CODE REVIEW

**Comment by hroest** [comment]
lets see whether this helps with travis and #1924

**Comment by hroest** [comment]
@cbielow maybe you can comment on this as you have used it in the past

**Comment by timosachsenberg** [comment]
am I correct that OpenMS.ini will not be created anymore?

File: src/openms/source/SYSTEM/File.cpp [hunk 1]
```diff
@@ -478,32 +478,9 @@ namespace OpenMS
     String filename = home_path + "/.OpenMS/OpenMS.ini";
 
     Param p;
-    if (!File::readable(filename)) // create file
+    if (!File::readable(filename)) // no file, lets keep it that way
```
Comment (hroest) [inline src/openms/source/SYSTEM/File.cpp]: no OpenMS.ini, dont try to create one

**Comment by hroest** [comment]
@timosachsenberg yes, that is correct. If it does not exist, dont try to create one (could lead to race conditions). This is not always a bad idea if you are on a temporary node or system where multiple threads try to create the file at the same time ...

**Comment by timosachsenberg** [comment]
In principle I agree. Is the folder still created? Otherwise the "lock" files for the update check can't be created.

**Comment by hendrikweisser** [comment]
> If it does not exist, dont try to create one (could lead to race conditions).

Isn't there an atomic operation to create a file only if it doesn't exist? Then you could try that without checking for existence beforehand.

**Comment by timosachsenberg** [comment]
afaik not for all file systems

**Comment by cbielow** [comment]
http://doc.qt.io/qt-5/qlockfile.html

We need Qt5.1 for that though...

**Comment by hroest** [comment]
it does not seem to help much with our issue on OSX

**Comment by timosachsenberg** [comment]
I am personally in favor of getting rid of automatic creation of OpenMS.ini

**Comment by hroest** [comment]
or at least we could do so for the tests, maybe using an ENV variable to disable it? 

we just run into issues again and again

http://cdash.openms.de/testDetails.php?test=33478083&build=185662
```
Warning: Cannot create '.OpenMS/OpenMS.ini' in user home directory. Please check your environment!
         Home directory determined is: /Users/travis.
Error: File not found (the file '/Users/travis/.OpenMS/OpenMS.ini' could not be found)
```

http://cdash.openms.de/testDetails.php?test=33305397&build=185662
```Error: Unable to read file (While loading '/Users/travis/.OpenMS/OpenMS.ini': invalid document structure( in line 1 column 1) in: /Users/travis/.OpenMS/OpenMS.ini)
/Users/travis/build/OpenMS/OpenMS/src/openms/source/FORMAT/HANDLERS/XMLHandler.cpp(103): While loading '/Users/travis/.OpenMS/OpenMS.ini': invalid document structure( in line 1 column 1)
```

its just annoying to have these test failures all the time.

**Comment by hroest** [comment]
@timosachsenberg any thoughts on this, do we accept this solution or continue looking?

**Comment by timosachsenberg** [comment]
I like the idea of not creating the ini file. Could you check if there are problems with the update check? It uses the same folder and we might create it first if the folder doesn't exist...

**Comment by hroest** [comment]
@timosachsenberg I suggest that people who run into problems with the update check can disable it easily through the setting of env variables.

secondly, I think the update checker fails more gracefully, see https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/SYSTEM/UpdateCheck.cpp#L94 line 94 and 106: if something fails and the file is not created, it will not readable and the block at line 106 should be skipped. So I dont think there will be problems. If there are, lets address them at that point.

**Comment by timosachsenberg** [comment]
If the file is not readable because it could not be created there will be no update check. This is not acceptable as this PR will basically disable the update check for all new users.

**Comment by timosachsenberg** [comment]
+1 seems to work. Thanks!

**Pull request merged**